src/java.base/share/classes/java/nio/file/Files.java

```diff
@@ -1569,7 +1569,8 @@ public static boolean isSameFile(Path path, Path path2) throws IOException {
      * {@code mismatch(f,f)} returns {@code -1L}). If the file system and files
      * remain static, then this method is <i>symmetric</i> (for two {@code Paths f}
      * and {@code g}, {@code mismatch(f,g)} will return the same value as
-     * {@code mismatch(g,f)}).
+     * {@code mismatch(g,f)}). If both paths locate the same file, the file
+     * system is not accessed.
```

Comment:
The only case where there is no file system access is when the paths are equal.

Reply:
Changed in 1bd9436ad7567b96c1baa14489721d93d7e7a436.